warehouse/oidc/models/gitlab.py

```diff
@@ -177,7 +177,20 @@ def __lookup_all__(klass, signed_claims: SignedClaims) -> Query | None:
                 environment=environment,
             )
             .filter(
-                literal(ci_config_ref).like(func.concat(klass.workflow_filepath, "%"))
+                literal(ci_config_ref).startswith(
+                    # See `__lookup_all__` in GitHubPublisherMixin for an
+                    # explanation of this mess.
+                    func.replace(
+                        func.replace(
+                            func.replace(klass.workflow_filepath, "\\", "\\\\"),
+                            "%",
+                            "\\%",
+                        ),
+                        "_",
+                        "\\_",
+                    ),
+                    escape="\\",
+                )
```

Comment:
Can we do the same here?

Reply:
Yep! The regexp is slightly different thanks to GitLab's support for path-qualified workflows, but I'll customize it.